plugin/discovery.go

```diff
@@ -117,15 +117,17 @@ func getPluginDir() (string, error) {
 // Only in the case of Windows, the pattern with the `.exe` is also considered,
 // and if it exists, the extension is added to the argument.
 func findPluginPath(path string) (string, error) {
-	_, err := os.Stat(path)
+	info, err := os.Stat(path)
 	if os.IsNotExist(err) && runtime.GOOS != "windows" {
 		return "", os.ErrNotExist
-	} else if !os.IsNotExist(err) {
+	} else if !os.IsNotExist(err) && !info.IsDir() {
```

Comment:
If you add a condition here, it will cause unintended behavior in the following cases:

- `runtime.GOOS == "linux"`
- path exists, but it's a directory

In this case, `findPluginPath` searches for the `.exe` file, which should only be handled on Windows.
In order to organize the implementation, it may be better to first branch the process according to the runtime OS.

Reply:
Hi @wata727, I've refactored it into a separate function to tidy up the intent and understanding.